skycoin/skywire
# Implement visor restart from hypervisor

Fixes #64 

 Changes:	
- Implement visor restart from hypervisor. It can be used via `/nodes/{pk}/restart` endpoint.

How to test this PR:

1. Unit tests
- Run added unit tests for `./pkg/restart/`

2. Integration tests
- Run the generic integration environment 
- Run `ps aux | grep visor`
- Check visors' PIDs
- In `shell` pane run `curl http://localhost:8080/api/nodes/024ec47420176680816e0406250e7156465e4531f5b26057c9f6297bb0303558c7/restart` 
- In `NodeA` pane make sure visor was restarted
- Run `ps aux | grep visor`
- Make sure that `NodeA`'s PID differs
